benchmarks/src/main/java/zipkin2/it/ServerIntegratedBenchmark.java

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright 2015-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.it;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.io.Closer;
+import com.linecorp.armeria.client.HttpClient;
+import io.netty.handler.codec.http.QueryStringEncoder;
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.utility.MountableFile;
+import zipkin.server.ZipkinServer;
+
+class ServerIntegratedBenchmark {
+
+  static final Logger LOGGER = LoggerFactory.getLogger(ServerIntegratedBenchmark.class);
+
+  static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  @Nullable static final String RELEASED_ZIPKIN_VERSION = System.getenv("ZIPKIN_VERSION");
+
+  static final boolean WAIT_AFTER_BENCHMARK = "true".equals(System.getenv("ZIPKIN_BENCHMARK_WAIT"));
+
+  Closer closer;
+
+  @BeforeEach void setUp() {
+    closer = Closer.create();
+  }
+
+  @AfterEach void tearDown() throws Exception {
```

Comment:
I would highly recommend to explicitly destroy the containers after the run. There is also an undocumented way to trigger the cleanup:
```java
ResourceReaper.instance().performCleanup();
```

Although Testcontainers will cleanup them later on, the deferred cleanup may affect the results of the next iteration

Reply:
The closer here is used to close the containers. Is `GenericContainer.stop` not enough to clean up?

Reply:
My bad, I didn't catch that the closer is for containers :) Yes, `stop`/`close` is enough 👍 

Although, the `ResourceReaper`-based suggestion still applies, and can simplify the thing (no need to register containers to `Closer`), but that's up to you, shouldn't make a lot of difference

Reply:
I tried using this but unfortnately it didn't work  -`ResourceReaper.instance().registeredContainers` is empty. Not sure if it's because of using `deepStart` or not but this would have been a nice shortcut given the parallel containers, sticking with manual for now though.